### PR TITLE
RavenDB-16963 - Using Group as a property name referencing a document prevent RavenDB from including it

### PR DESCRIPTION
**Pull request by grisha-kotler**
https://issues.hibernatingrhinos.com/issue/RavenDB-16963

### CODE REVIEW

File: src/Raven.Client/Documents/Subscriptions/DocumentSubscriptions.cs [hunk 1]
```diff
@@ -192,7 +193,7 @@ public string Create<T>(SubscriptionCreationOptions<T> options, string database
                         if (IncludesUtil.RequiresQuotes(include, out var escapedInclude))
                             criteria.Query += $"'{escapedInclude}'";
                         else
-                            criteria.Query += include;
+                            criteria.Query += QueryToken.IsKeyword(include) ? $"'{include}'" : include;
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/DocumentSubscriptions.cs:196]: What if the field contains `'` alredy?

Comment (grisha-kotler) [reply src/Raven.Client/Documents/Subscriptions/DocumentSubscriptions.cs:196]: It is handled here: https://github.com/ravendb/ravendb/blob/v5.2/src/Raven.Client/Documents/Session/IncludesUtil.cs#L140

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs [hunk 2]
```diff
@@ -8,9 +8,9 @@ public abstract class QueryToken
     {
         public abstract void WriteTo(StringBuilder writer);
 
-        protected  void WriteField(StringBuilder writer, string field)
+        public static void WriteField(StringBuilder writer, string field)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Tokens/QueryToken.cs]: internal

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs [hunk 3]
```diff
@@ -21,7 +21,12 @@ protected  void WriteField(StringBuilder writer, string field)
                 writer.Append("'");
         }
 
-        private static readonly HashSet<string> RqlKeywords = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
+        public static bool IsKeyword(string field)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Tokens/QueryToken.cs]: internal

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs [hunk 4]
```diff
@@ -21,7 +21,12 @@ protected  void WriteField(StringBuilder writer, string field)
...
+        {
+            return RqlKeywords.Contains(field);
+        }
+
+        internal static readonly HashSet<string> RqlKeywords = new HashSet<string>(StringComparer.OrdinalIgnoreCase)
```
Comment (ppekrol) [inline src/Raven.Client/Documents/Session/Tokens/QueryToken.cs:29]: Does it have to be internal? Can be private?

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs (hunk 2 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/Tokens/QueryToken.cs]: done

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs (hunk 3 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/Tokens/QueryToken.cs]: done

File: src/Raven.Client/Documents/Session/Tokens/QueryToken.cs (hunk 4 above)
Comment (grisha-kotler) [reply src/Raven.Client/Documents/Session/Tokens/QueryToken.cs:29]: I'm using it in a test https://github.com/ravendb/ravendb/pull/12475/files#diff-381b92fee1fd6309a09f66171eb7aa8140b083104029396a99b5a629157892d8R31

**Pull request merged**